app/controllers/users_controller.rb

```diff
@@ -34,37 +34,56 @@ def edit
   end
 
   def update
```

Comment:
this update method is too long and has too many nested conditionals in my opinion. i created a refactoring proposal in the PR #398 I will add comments here anyway in order to explain the reasoning behind this refactoring

Reply:
(comments in the Refactoring PR)